plugin.go

```diff
@@ -385,9 +385,9 @@ func (d plugin) getByName(name string) (*volumes.Volume, error) {
 	return volume, err
 }
 
-func (d plugin) detachVolume(ctx context.Context, vol *volumes.Volume) (*volumes.Volume, error) {
```

Comment:
How about we keep `context.Context` as an argument to all internal functions, and pass a single `context.TODO()` from the main functions?

```
func (d plugin) Mount(r *volume.MountRequest) (*volume.MountResponse, error) {
    ctx := context.TODO()
    ...
        if vol, err = d.waitOnVolumeState(ctx, vol, "available"); err != nil {
    ...
        if vol, err = d.detachVolume(ctx, vol); err != nil {
    ...
```

This way, we would be able to e.g. pass a cancelable context later if needed (https://pkg.go.dev/context).

Reply:
That's a good idea. Should the same instance of `context.TODO()` be passed to the other functions using their own instance of it at the moment?

Like in this line, for example:
https://github.com/SLH335/docker-plugin-cinder/blob/93cab39892abd699b64a6b3f67d38d5be2bed382/plugin.go#L192

Reply:
Yes, that sounds about right! 👍